kraft-jennifer-ndvi-automation.py

```diff
@@ -0,0 +1,698 @@
+        
+        mean_ndvi = ndvi.mean(skipna=True)                                #get mean! yay. Also skip all na values
+        
+        dataContainer.append( (siteName, date, mean_ndvi.item()) )        #append tuple of results to dataContainer!
+        
+DF = pd.DataFrame(dataContainer,                                          #Create dataframe from list of tuples
+            columns=["site","date","mean_ndvi"])                          #with columns site, date, mean_ndvi
+
+DF["date"] = pd.to_datetime(DF["date"], format="%Y%m%d")                  #convert string rep of date col to datetime obj
+DF.set_index("date", inplace=True)                                        #set index of df to also be datetime
+
+DF
+
+
+# In[10]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points +=2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points +=2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points +=3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points +=3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# In[11]:
+
+
+# Add only the plot code to this cell
+
+# This is the final figure of mean NDVI
+# for both sites across the year
+# with data cleaned to deal with clouds
+
+# YOUR CODE HERE
+fig, ax = plt.subplots(figsize=(15,10))
+
+ax.set_xlabel("Date")
+ax.set_ylabel("Mean NDVI")
+ax.set_title("Mean Normalized Difference Vegetation Index (NDVI)\n Jan 2017 - Dec 2017\n Landsat 8 with Cloud Removed")
+
+DF.dropna(inplace=True)
```

Comment:
`dropna` seems like another great tool. I don't remember why now, but I had a hard time getting rid of the NA's in my DataFrame during this assignment.

Reply:
I had a lot of trouble with that at first too. I wish I could remember where I dug this up, but I was very happy to discover it.